test/expected-output-test/multiple_rules_for_same_head/compile.expected

```diff
@@ -342,7 +342,7 @@ function: """Imply(Q.R0.dd_label, Q.R1.dd_label)"""
 
 }
 
-deepdive.inference.factors.inf2_imply_Q_Q {
+deepdive.inference.factors.inf_imply_Q_Q2 {
```

Comment:
Let's add another rule in the test `Q(x) => Q1(x) :- ...` with a new variable `Q1` to make sure our unique name generation is robust.  Probably we want a similar test for `@name()` as well.